File: composeApp/src/desktopMain/kotlin/org/queststudios/yamlvalidation/validation/impl/TypologyValidation.kt
package org.queststudios.yamlvalidation.validation.impl

import org.queststudios.yamlvalidation.core.ValidationContext
import org.queststudios.yamlvalidation.i18n.Strings
import org.queststudios.yamlvalidation.validation.ValidationLogger
import org.queststudios.yamlvalidation.validation.ValidationRule

class TypologyValidation : ValidationRule {
    override val name: String = "TypologyValidation"

    override fun validate(endpoint: String, method: String, context: ValidationContext, logger: ValidationLogger) {
        val language = org.queststudios.yamlvalidation.core.AppConfig.language
        try {
            val yamlData = context.yamlData
            val paths = yamlData["paths"] as? Map<*, *> ?: return
            val endpointObj = paths[endpoint] as? Map<*, *> ?: return
            val methodObj = endpointObj[method] as? Map<*, *> ?: return
            val typologyBlock = methodObj["x-typology"] as? Map<*, *> ?: return
            val typology = typologyBlock["typology"]?.toString() ?: ""
            if (typology != "external") {
                logger.log("ERROR", Strings.get(language, "typology.error.not_external").replace("{0}", typology))
            } else {
                logger.log("SUCCESS", Strings.get(language, "typology.success.external"))
            }
        } catch (e: Exception) {
            logger.log("ERROR", "Error en validarTypology: ${e.message}")
        }
    }
}

File: src/validation/rules/TypologyValidation.js
import { ValidationRule } from '../ValidationRule.js';
import { Strings } from '../../i18n/strings.js';
import { AppConfig } from '../../core/AppConfig.js';

export class TypologyValidation extends ValidationRule {
  validate(endpoint, method, context, logger) {
    const language = AppConfig.language;
    try {
      const yamlData = context.yamlData;
      const paths = yamlData.paths || {};
      const endpointObj = paths[endpoint];
      if (!endpointObj) return;
      
      const methodObj = endpointObj[method];
      if (!methodObj) return;
      
      const typologyBlock = methodObj['x-typology'];
      if (!typologyBlock) return;
      
      const typology = typologyBlock.typology || '';
      
      if (typology !== 'external') {
        logger.log('ERROR', Strings.get(language, 'typology.error.not_external', typology));
      } else {
        logger.log('SUCCESS', Strings.get(language, 'typology.success.external'));
      }
    } catch (error) {
      logger.log('ERROR', `Error en validarTypology: ${error.message}`);
    }
  }
}

